src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,249 @@
+        `/api/v2/custom_objects/${customObjectKey}/records/${selectionValue}`
+      );
+      const { custom_object_record } = await res.json();
+      const newSelectedOption = {
+        name: custom_object_record.name,
+        value: custom_object_record.id,
+      };
+      setSelectedOption(newSelectedOption);
+      setInputValue(custom_object_record.name);
+      onChange(custom_object_record.id);
+    },
+    [customObjectKey, onChange]
+  );
+
+  const handleChange = useCallback<NonNullable<IComboboxProps["onChange"]>>(
+    async ({ inputValue, selectionValue }) => {
+      setIsFirstLoad(false);
+      if (selectionValue !== undefined) {
+        if (selectionValue == "") {
+          setSelectedOption(EMPTY_OPTION);
+          setInputValue(EMPTY_OPTION.name);
+          onChange(EMPTY_OPTION.value);
+        } else {
+          await fetchSelectedOption(selectionValue as string);
+        }
+        return;
+      }
+
+      if (inputValue !== undefined) {
+        setInputValue(inputValue);
+        if (inputValue === "") {
+          setOptions([]);
+          setSelectedOption(null);
+          onChange("");
+        } else {
+          const searchParams = new URLSearchParams();
+          const organization_id = organizationId
+            ? organizationId
+            : await getOrganizationId(userId);
+          searchParams.set("name", inputValue.toLocaleLowerCase());
+          searchParams.set("source", "zen:ticket");
+          searchParams.set("field_id", fieldId.toString());
+          searchParams.set("user_id", userId.toString());
+          searchParams.set("organization_id", organization_id);
+          setIsLoadingOptions(true);
+          try {
+            const response = await fetch(
+              `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+            );
+
+            const data = await response.json();
+            if (data !== undefined) {
+              setOptions(
+                data.custom_object_records.map(
+                  ({ name, id }: { name: string; id: string }) => ({
+                    name,
+                    value: id,
+                  })
+                )
+              );
+            } else {
+              setOptions([]);
+            }
+          } catch (error) {
+            return error;
+          } finally {
+            setIsLoadingOptions(false);
+          }
+        }
+      }
+      return;
+    },
+    [
+      customObjectKey,
+      fetchSelectedOption,
+      userId,
+      organizationId,
+      fieldId,
+      onChange,
+    ]
+  );
+
+  const debounceHandleChange = useMemo(() => debounce(handleChange, 300), []);
+
+  useEffect(() => {
+    return () => debounceHandleChange.cancel();
+  }, [debounceHandleChange]);
+
+  useEffect(() => {
+    if (value && !options.find((option) => option.value === value)) {
+      fetchSelectedOption(value as string);
+      return;
+    }
+  }, []);
```

Comment:
Reasoning again about this, I think we should simplify things a bit.

This component calls `onChange` when the value changes, and then it will receive again the new `value` from the props from the main component. If I understand correctly, we want to set the `selectedOption` when the user selects an option or at the first render if we have a value.

I think we have three different ways of handling it.

**Option 1**
- As mentioned in the previous comment, in `handleChange` we can search for the selected option in the `options` array,  set the `selectedOption` as the option found, and then call `onChange` and `setInputValue` directly skipping `fetchSelectedOption`
- This `useEffect` can remain with `[]` as a dependency because we want to call `fetchSelectedOption` only on the first render

**Option 2**
- We keep things as they are now, but we move `onChange(custom_object_record.id);` in `handleChange` as `onChange(selectionValue as string);`
  - With the current implementation, on the first render we get a `value` -> we call `fetchSelectedOption` -> we emit an `onChange` with the same value. We can skip this unneeded step

**Option 3**
- We change this `useEffect` to have `[value]` as the dependency array, so it will call `fetchSelectedOption` every time the `value` changes
- We change `handleChange` to emit only `onChange` when an option is selected, then when the component re-renders it will call `fetchSelectedOption` to fetch the option
- We remove the `onChange` call from `fetchSelectedOption`

Option 3 is the cleanest code-wise, we just react with an effect to changes of the `value`, but it is not efficient, because when the user selects an option the component will re-render, we will call the API, and the component will re-render again, so I would not go for that one.

Option 1 is the most efficient because we avoid an extra HTTP call. Option 2 can be ok as well, especially if we are not confident that we always have the selected option in the `options` array